### ISSUE TITLE
Pre-training on cifar10 ends extremely fast...

### ISSUE THREAD

**Issue by Jayis** [google-research/simclr#92]
Hi,
I try to pre-train on cifar10 then linear-eval on cifar10.
But the pre-training ends within 10 mins...
It seems to be way too fast for training 1000 epochs on cifar10 using Resnet-18.

I try to find out the reason.
And it seems that you update the **global_step** multiple times in the **apply_gradients**() in LARS optimizer.
- https://github.com/google-research/simclr/blob/master/lars_optimizer.py#L140

Shouldn't the **global_step** only increased by 1 after you called the **apply_gradients**()?

---- Reproduce Detail -----
I simply run the commands you provide in the README.md.
- https://github.com/google-research/simclr/blame/master/README.md#L78
- https://github.com/google-research/simclr/blame/master/README.md#L115

**Comment by chentingpc** [comment]
This looks like a bug that was introduced in a recent commit (https://github.com/google-research/simclr/commit/180b56a9af72725c3ea694556782aac036f2935b). Can you try to move L140-L142 (the block of ```if global_step is not None:```) two spaces to the left, and see if this problem still exists? PS, you may want to use a new result folder so it won't automatically continue training.

**Comment by Jayis** [comment]
Moving the code block did seem to solve the problem.
Pre-training time now becomes 4~5 hrs, and linear evaluation now reaches accuracy around 91%.

Just want to make sure if that was a bug, since I'm new to tensorflow.
Thanks for your reply!

**Comment by chentingpc** [comment]
thanks for the update!

closing this issue as the bug has been fixed in https://github.com/google-research/simclr/commit/036ea7baa3eeb255c88a08392fcdc1d55bb8133f.
